amd/src/expand-topics.js

```diff
@@ -0,0 +1,48 @@
+const waitForElements = () => {
+    const courseIndexButtons = document.querySelectorAll('.courseindex-section');
+
+    if (courseIndexButtons.length > 0) {
+        // Add listeners.
+        courseIndexButtons.forEach(function (button) {
+            button.addEventListener('click', handleButtonClick);
+        });
+    } else {
+        // Elements not loaded yet.
+        setTimeout(waitForElements, 250);
```

Comment:
You need to use event delegation instead of adding a timeout and hoping that the buttons will be available

I can show you tomorrow how to do it if you need help